bindings/src/test/java/io/pravega/storage/filesystem/FileSystemChunkStorageMockTest.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.storage.filesystem;
+
+import io.pravega.segmentstore.storage.chunklayer.ChunkHandle;
+import lombok.Getter;
+import lombok.Setter;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.mockito.ArgumentCaptor;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.channels.FileChannel;
+import java.nio.channels.spi.AbstractInterruptibleChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.Executor;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Unit tests for {@link FileSystemChunkStorage} that uses mocks.
+ */
+public class FileSystemChunkStorageMockTest {
```

Comment:
I'm not sure I understand the purpose of this mock, there is only a single test case it is serving, `doReadTest`.

Reply:
There are some corner cases in read logic that can only be tested using Mockito mock.

Reply:
Ok, this name confused me. if this is about testing the read logic in a given way, then name it accordingly. I thought you're defining a mock to be used across test cases, while it sounds like there are aspects of the read logic that require mocking, so the test is about read, not mocking.

Reply:
This is not only about the read logic, but testing FileSystemChunkStorage functionality by mocking underlying file system apis.  In this context the name correctly reflects the intent.
Mocks are useful to create hard to reproduce scenario. In future I do intent to add more tests around handling exception.